addons/website_event_crm/models/event_registration.py

```diff
@@ -26,6 +26,33 @@ def _get_lead_description_fields(self):
         res.append('registration_answer_ids')
         return res
 
+    def _get_lead_grouping(self, rules, rule_to_new_regs):
+        visitor_registrations = self.filtered('visitor_id')
+        # We don't want to group by visitor if there is a sale order,
+        # this is done here to avoid another bridge module.
+        if 'sale_order_id' in self._fields:
+            visitor_registrations -= visitor_registrations.filtered('sale_order_id')
+        grouping_res = super(EventRegistration, self - visitor_registrations)._get_lead_grouping(rules, rule_to_new_regs)
+
+        if visitor_registrations:
+            related_leads = self.env['crm.lead'].search([
```

Comment:
in `_run_on_registration` we also include inactive leads, I was wondering if we should align ourselves here?